lua/entities/gmod_wire_textentry.lua

```diff
@@ -164,23 +171,36 @@ net.Receive("wire_textentry_action",function(len,ply)
 	self:Unprompt() -- in all cases, make text entry available for use again
 
 	if ok and not self.BlockInput then
-		WireLib.TriggerOutput( self, "Text", text )
+		self:OnTextEntered(text)
 
-		local timername = "wire_textentry_" .. self:EntIndex()
-		timer.Remove( timername )
-		if math.max(self:GetHold(),0) > 0 then
-			timer.Create( timername, math.max(self:GetHold(),0), 1, function()
-				if IsValid( self ) then
-					WireLib.TriggerOutput( self, "User", nil )
-					WireLib.TriggerOutput( self, "Text", "" )
-				end
-			end)
-		end
+		
 	end
 
 	self:UpdateOverlay()
 end)
 
+local ENTERED_DISABLE_DELAY = 0.1
+
+function ENT:OnTextEntered(text)
+	WireLib.TriggerOutput( self, "Text", text )
+	WireLib.TriggerOutput( self, "Entered", 1 )
+
+	local timername = "wire_textentry_" .. self:EntIndex()
+	timer.Remove( timername )
+	if math.max(self:GetHold(),0) > 0 then
+		timer.Create( timername, math.max(self:GetHold(),0), 1, function()
```

Comment:
Store `math.max(self:GetHold(), 0)` in a variable describing what it is (and to reuse it)